src/CoreDumpWriter.c

```
@@ -300,6 +300,10 @@ FILE *popen2(const char *command, const char *type, pid_t *pid)
 // remove all non alphanumeric characters from process name and replace with '_'
 char *sanitize(char * processName)
 {
+    if(processName == NULL){
```

Since this change makes `GetProcessName` always return a value, I think we should check for `null`, but instead just error out rather than return `null` again.

---

But I thought this is for disaster situation that you mentioned before.

---

Due to the modifications you are making to `GetProcessName` it should never return a `NULL` value. Due to this, if for some disastrous reason `sanitize` is passed a `NULL` it should error out and exit Procdump rather then returning `NULL` leading towards a potential segfault when writing the dump file name in `WriteCoreDumpInternal`.